src/include/planner/logical_plan/copy/logical_copy_from.h

```diff
@@ -10,53 +10,45 @@ namespace planner {
 class LogicalCopyFrom : public LogicalOperator {
 
 public:
-    LogicalCopyFrom(const common::CopyDescription& copyDescription, common::table_id_t tableID,
-        std::string tableName, bool hasSerial, binder::expression_vector dataColumnExpressions,
+    LogicalCopyFrom(std::unique_ptr<common::CopyDescription> copyDescription,
+        catalog::TableSchema* tableSchema, binder::expression_vector columnExpressions,
         std::shared_ptr<binder::Expression> nodeOffsetExpression,
-        std::shared_ptr<binder::Expression> outputExpression)
+        std::shared_ptr<binder::Expression> outputExpression, bool preservingOrder_)
         : LogicalOperator{LogicalOperatorType::COPY_FROM},
-          copyDescription{copyDescription}, tableID{tableID}, tableName{std::move(tableName)},
-          preservingOrder{hasSerial}, dataColumnExpressions{std::move(dataColumnExpressions)},
-          nodeOffsetExpression{std::move(nodeOffsetExpression)}, outputExpression{
-                                                                     std::move(outputExpression)} {}
+          copyDescription{std::move(copyDescription)}, tableSchema{tableSchema},
+          columnExpressions{std::move(columnExpressions)}, nodeOffsetExpression{std::move(
+                                                               nodeOffsetExpression)},
+          outputExpression{std::move(outputExpression)}, preservingOrder_{preservingOrder_} {}
 
-    inline std::string getExpressionsForPrinting() const override { return tableName; }
-
-    inline common::CopyDescription getCopyDescription() const { return copyDescription; }
-
-    inline common::table_id_t getTableID() const { return tableID; }
-
-    inline std::vector<std::shared_ptr<binder::Expression>> getDataColumnExpressions() const {
-        return dataColumnExpressions;
-    }
+    inline std::string getExpressionsForPrinting() const override { return tableSchema->tableName; }
 
+    inline common::CopyDescription* getCopyDescription() const { return copyDescription.get(); }
+    inline catalog::TableSchema* getTableSchema() const { return tableSchema; }
+    inline binder::expression_vector getColumnExpressions() const { return columnExpressions; }
     inline std::shared_ptr<binder::Expression> getNodeOffsetExpression() const {
         return nodeOffsetExpression;
     }
-
     inline std::shared_ptr<binder::Expression> getOutputExpression() const {
         return outputExpression;
     }
 
-    inline bool isPreservingOrder() { return preservingOrder; }
+    inline bool preservingOrder() { return preservingOrder_; }
```

Comment:
Consider renaming to `orderPreserving` or `preserveOrder`, ditto on the field `preservingOrder_`.